plugin/keyring_vault/vault_io.h

```diff
@@ -0,0 +1,54 @@
+#ifndef MYSQL_VAULT_IO_H
+#define MYSQL_VAULT_IO_H
+
+#include <my_global.h>
+#include <logger.h>
+#include "i_vault_io.h"
+#include "vault_parser.h"
+#include "vault_curl.h"
+#include "vault_key_serializer.h"
+
+namespace keyring {
+
+class Vault_io : public IVault_io
+{
+public:
+  Vault_io(ILogger *logger, IVault_curl *vault_curl,
```

Comment:
I think this is something you would need to argue with Google, Oracle and possibly Stroustrup about. Apparently they all think (and me too) that passing by reference is last resort. in your getline example you cannot tell if is or str is modified by getline. So the function name does not help and it is not so ofter that you can function this way. Anyways I do not think we should go into this further. I am OK with you using references and me trying to stick to const references only.